flink/v1.13/flink/src/test/java/org/apache/iceberg/flink/TestFlinkUpsert.java

```diff
@@ -210,33 +189,7 @@ public void testPrimaryKeyFieldsAtBeginningOfSchema() {
               "PARTITIONED BY (data) WITH %s",
           tableName, toWithClause(tableUpsertProps));
 
-      sql("INSERT INTO %s VALUES " +
-          "('aaa', DATE '2022-03-01', 1)," +
-          "('aaa', DATE '2022-03-01', 2)," +
-          "('bbb', DATE '2022-03-01', 3)",
-          tableName);
-
-      TestHelpers.assertRows(
-          sql("SELECT * FROM %s", tableName),
-          Lists.newArrayList(Row.of("aaa", dt, 2), Row.of("bbb", dt, 3)));
-
-      sql("INSERT INTO %s VALUES " +
-          "('aaa', DATE '2022-03-01', 4)," +
-          "('bbb', DATE '2022-03-01', 5)",
-          tableName);
-
-      TestHelpers.assertRows(
-          sql("SELECT * FROM %s", tableName),
-          Lists.newArrayList(Row.of("aaa", dt, 4), Row.of("bbb", dt, 5)));
-
-      sql("INSERT INTO %s VALUES " +
-          "('aaa', DATE '2022-03-01', 6)," +
-          "('bbb', DATE '2022-03-01', 7)",
-          tableName);
-
-      TestHelpers.assertRows(
-          sql("SELECT * FROM %s", tableName),
-          Lists.newArrayList(Row.of("aaa", dt, 6), Row.of("bbb", dt, 7)));
+      testUpsert(tableName, row -> Row.of(row.getField(1), dt, row.getField(0)));
```

Comment:
In my experience, when tests have failed, when of the indicators is that one or more of the resulting `Row`s will have extra fields.

So is mapping into specific Row fields possibly making it more likely that we will miss those things?